pytensor/tensor/basic.py

```diff
@@ -760,7 +760,31 @@ def switch(cond, ift, iff):
     """if cond then ift else iff"""
 
 
-where = switch
+def where(cond, ift=None, iff=None, **kwargs):
+    """
+    where(condition, [ift, iff])
+    Return elements chosen from `ift` or `iff` depending on `condition`.
+
+    Note: When only condition is provided, this function is a shorthand for `as_tensor(condition).nonzero()`.
+
+    Parameters
+    ----------
+    condition : tensor_like, bool
+        Where True, yield `ift`, otherwise yield `iff`.
+    x, y : tensor_like
+        Values from which to choose.
+
+    Returns
+    -------
+    out : TensorVariable
+        A tensor with elements from `ift` where `condition` is True, and elements from `iff` elsewhere.
+    """
+    if ift is not None and iff is not None:
+        return switch(cond, ift, iff)
+    elif ift is None and iff is None:
+        return as_tensor(cond).nonzero()
```

Comment:
Sorry I missed this. We should propagate the kwargs
```suggestion
        return switch(cond, ift, iff, **kwargs)
    elif ift is None and iff is None:
        return as_tensor(cond).nonzero(**kwargs)
```